packages/agent/src/agent/http/index.ts

```diff
@@ -194,7 +206,12 @@ export class HttpAgent implements Agent {
     },
     identity?: Identity | Promise<Identity>,
   ): Promise<SubmitResponse> {
-    const id = (await (identity !== undefined ? await identity : await this._identity)) as Identity;
+    const id = await (identity !== undefined ? await identity : await this._identity);
```

Comment:
Why is `_identity` a Promise?

Reply:
Decisions that predate me - @hansl liked to pre-emptively make API's compatible with async / await, even if it was not required at time of implementation